include/exadg/poisson/spatial_discretization/operator.cpp

```diff
@@ -131,7 +131,11 @@ Operator<dim, n_components, Number>::distribute_dofs()
 
   dof_handler.distribute_dofs(*fe);
 
-  // affine constraints only relevant for continuous FE discretization
+  // Affine constraints are only relevant for continuous Galerin discretization.
+  // The AffineConstraints object is used to initialize MatrixFree. Here, we apply homogeneous
+  // boundary conditions as needed by vmult() in iterative solvers for linear systems of equations,
+  // implemented via dealii::MatrixFree and FEEvaluation::read_dof_values() (or gather_evaluate()).
+  // The actual inhomogeneous boundary data needs to be imposed separately and explicitly in ExaDG.
```

Comment:
I think this is a good summary, but it sounds a bit like it would be ExaDG's fault. It is rather a design decision (or general limitation) in the implementation of `MatrixFree` and the related evaluators `FEEvaluation` in deal.II.

Reply:
Suggestion: "// The actual inhomogeneous boundary data needs to be imposed separately due to the implementation in deal.II that can not handle multiple AffineConstraints. Hence, we need to do this explicitly in ExaDG."

What do you think about this version, @kronbichler?

Reply:
Yes, that is good.